ocaml/sdk-gen/go/gen_go_helper.ml

```diff
@@ -242,40 +242,94 @@ module Json = struct
             ; ("func_name_suffix", `String (suffix_of_type t))
             ]
 
-  let of_params params =
-    let name_internal name =
-      let name = name |> snake_to_camel |> String.uncapitalize_ascii in
-      match name with "type" -> "typeKey" | "interface" -> "inter" | _ -> name
+  let rec get_last = function
+    | [] ->
+        None
+    | [x] ->
+        Some x
+    | _ :: tl ->
+        get_last tl
+
+  let group_params msg =
+    let raw_params =
+      if msg.msg_session then
+        session_id :: msg.msg_params
+      else
+        msg.msg_params
     in
-    let of_param param =
-      let suffix_of_type = suffix_of_type param.param_type in
-      let t, _e = string_of_ty_with_enums param.param_type in
-      let name = param.param_name in
-      [
-        ("is_session_id", `Bool (name = "session_id"))
-      ; ("type", `String t)
-      ; ("name", `String name)
-      ; ("name_internal", `String (name_internal name))
-      ; ("doc", `String param.param_doc)
-      ; ("func_name_suffix", `String suffix_of_type)
-      ]
+    let params =
+      List.map
+        (fun p -> (get_published_release_for_param p.param_release.internal, p))
+        raw_params
+    in
+    let uniq_sorted_rels =
+      List.map fst params
+      |> Listext.List.setify
+      |> List.fast_sort compare_versions
     in
-    (* We use ',' to seprate params in Go function, we should ignore ',' before first param,
-       for example `func(a type1, b type2)` is wanted rather than `func(, a type1, b type2)`.
-    *)
-    let add_first = function
-      | head :: rest ->
-          let head = `O (("first", `Bool true) :: of_param head) in
-          let rest =
-            List.map
-              (fun item -> `O (("first", `Bool false) :: of_param item))
-              rest
+    let groups =
+      List.map
+        (fun rel ->
+          let params =
+            List.filter_map
+              (fun (r, param) ->
+                match compare_versions r rel with
+                | n when n > 0 ->
+                    None
+                | _ ->
+                    Some param
+              )
```

Comment:
```
List.filter (fun (r, _) -> (compare_version r rel) > 0)) params
```

Reply:
Then we need to:
`List.map snd params`. Then it equals `List.filter_map`